### PR TITLE
[IMP] add contectless payment lines reconcile feature

### PR DESCRIPTION
**Pull request by siddharth7815**
Task Link :

https://www.druidoo.io/web?db=druidoo-druidoo-master-349765&token=J39sYB1yfc1NMuVYgIBr#id=1080&action=367&model=project.task&view_type=form&menu_id=237

### CODE REVIEW

**Comment by ivantodorovich** [comment]
travis is failing because of LINT:

```
account_reconcile_pos_payments/models/account_bank_statement.py:8:1: F401 'datetime.datetime' imported but unused
account_reconcile_pos_payments/models/account_bank_statement.py:85:89: E501 line too long (96 > 88 characters)
account_reconcile_pos_payments/models/account_bank_statement.py:103:89: E501 line too long (90 > 88 characters)
account_reconcile_pos_payments/models/account_bank_statement.py:120:89: E501 line too long (94 > 88 characters)
account_reconcile_pos_payments/models/account_bank_statement.py:146:89: E501 line too long (103 > 88 characters)
account_reconcile_pos_payments/models/account_bank_statement.py:184:89: E501 line too long (95 > 88 characters)
account_reconcile_pos_payments/models/account_bank_statement.py:185:89: E501 line too long (105 > 88 characters)
account_reconcile_pos_payments/models/account_bank_statement.py:192:89: E501 line too long (91 > 88 characters)
account_reconcile_pos_payments/models/account_bank_statement.py:223:89: E501 line too long (102 > 88 characters)
account_reconcile_pos_payments/models/account_bank_statement.py:230:89: E501 line too long (103 > 88 characters)
```

File: account_reconcile_pos_payments/models/account_bank_statement.py [hunk 1]
```diff
@@ -210,28 +214,26 @@ def process_statement_reconcile(self, lines, statement_id, ct_line=False):
...
+                        statement_id.journal_id.default_credit_account_id.id
+                        and move_line.id not in lines_to_reconcile):
                     lines_to_reconcile.append(move_line.id)
 
-            move_lines = self.env['account.move.line'].browse(
-                lines_to_reconcile)
-            move_lines.reconcile()
+        move_lines = self.env['account.move.line'].browse(
+            lines_to_reconcile)
+        move_lines.reconcile()
```
Comment (ivantodorovich) [inline account_reconcile_pos_payments/models/account_bank_statement.py:243]: So, the only actual change is in the indentation of these 3 lines, right?

File: account_reconcile_pos_payments/models/account_bank_statement.py [hunk 2]
```diff
@@ -210,28 +214,26 @@ def process_statement_reconcile(self, lines, statement_id, ct_line=False):
...
                     lines_to_reconcile.append(move_line.id)
 
             for move_line in line.journal_entry_ids:
-                if (
-                        move_line.account_id.id == statement_id.journal_id.default_credit_account_id.id
-                        and move_line.id not in lines_to_reconcile
-                ):
+                if (move_line.account_id.id ==
+                        statement_id.journal_id.default_credit_account_id.id
+                        and move_line.id not in lines_to_reconcile):
```
Comment (ivantodorovich) [inline account_reconcile_pos_payments/models/account_bank_statement.py]: Please avoid changing lines if they're not necessary
Also .. yes , maybe the line was too long before, but it's actually much more readable

If the line is too long you can do this:

```suggestion
                default_debit_account_id = statement_id.journal_id.default_debit_account_id.id
                if (
                    move_line.account_id.id == default_debit_account_id
                    and move_line.id not in lines_to_reconcile
                ):
```

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

**Comment by ivantodorovich** [comment]
@siddharth7815 can you finish this PR please?

File: account_reconcile_pos_payments/models/account_bank_statement.py (hunk 1 above)
Comment (siddharth7815) [reply account_reconcile_pos_payments/models/account_bank_statement.py:243]: @ivantodorovich .. yes

File: account_reconcile_pos_payments/models/account_bank_statement.py (hunk 2 above)
Comment (siddharth7815) [reply account_reconcile_pos_payments/models/account_bank_statement.py]: @ivantodorovich .. it's done as per your suggestion.

**Pull request merged**